dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/response/SolrSearchResult.java

```diff
@@ -933,72 +881,37 @@ public String getFileParentIdentifier() {
         if (entity == null) {
             return null;
         }
-        if (entity instanceof DataFile) {
-            return parent.get(PARENT_IDENTIFIER);   // Dataset globalID
+        if (type == SearchObjectType.FILES) {
+            return parent.getParentIdentifier();   // Dataset globalID
         }
 
         return null;
-        //if (entity)
-    }
-
-    public String getFilePersistentId() {
-        return filePersistentId;
     }
 
     public void setFilePersistentId(String pid) {
         filePersistentId = pid;
     }
 
     public String getFileUrl() {
-        // Nothing special needs to be done for harvested file URLs:
-        // simply directing these to the local dataset.xhtml for this dataset
-        // will take care of it - because DatasetPage will issue a redirect
-        // to the remote archive URL.
-        // This is true AS OF 4.2.4, FEB. 2016! - We'll probably want to make
-        // .getRemoteArchiveURL() methods, both in DataFile and Dataset objects,
-        // work again at some point in the future.
-        /*
-        if (entity != null && entity instanceof DataFile && this.isHarvested()) {
-            String remoteArchiveUrl = ((DataFile) entity).getRemoteArchiveURL();
-            if (remoteArchiveUrl != null) {
-                return remoteArchiveUrl;
-            }
-            return null;
-        }*/
-        if (entity.getIdentifier() != null) {
+        if (identifier != null) {
             if (isDraftState()) {
-                return "/file.xhtml?persistentId=" + entity.getGlobalIdString() + "&version=DRAFT";
+                return "/file.xhtml?persistentId=" + identifier + "&version=DRAFT";
```

Comment:
We probably will want to slightly modify indexing on this one:
`datafileSolrInputDocument.addField(SearchFields.IDENTIFIER, fileEntityId);` in `IndexServiceBean` should index globalId instead of entityId